rho/scancommand.py

```diff
@@ -50,57 +49,46 @@ def _create_ping_inventory(vault, vault_pass, profile_ranges, profile_port,
         # pylint: disable=anomalous-backslash-in-string
         reg = "[0-9]*.[0-9]*.[0-9]*.\[[0-9]*:[0-9]*\]"
         profile_range = profile_range.strip(',').strip()
+        hostname = str_to_ascii(profile_range)
         if not re.match(reg, profile_range):
-            string_to_write += profile_range + \
-                " ansible_host=" + profile_range + " ansible_port=" + \
-                str(profile_port) + "\n"
+            hosts_dict[profile_range] = {'ansible_host': profile_range,
+                                         'ansible_port': profile_port}
         else:
-            string_to_write += profile_range + "\n"
-
-    string_to_write += '\n'
-
-    string_header = copy(string_to_write)
+            hosts_dict[hostname] = None
 
+    vars_dict = dict()
     for cred_item in profile_auth_list:
+        cred_pass = cred_item.get('password')
+        cred_sshkey = cred_item.get('ssh_key_file')
         auth_item = [cred_item.get('id'),
                      cred_item.get('name'),
                      cred_item.get('username'),
                      cred_item.get('password'),
                      cred_item.get('ssh_key_file')]
 
-        ping_inventory = open('data/ping-inventory', 'w')
-        string_to_write = \
-            string_header + \
-            "[all:vars]\n" + \
-            "ansible_user=" + \
-            auth_item[2]
-
-        auth_pass_or_key = ''
+        vars_dict['ansible_user'] = str_to_ascii(cred_item.get('username'))
 
-        if (not auth_item[3] == '') and auth_item[3]:
-            auth_pass_or_key = '\nansible_ssh_pass=' + auth_item[3]
-            if (not auth_item[4] == '') and auth_item[4]:
-                auth_pass_or_key += "\nansible_ssh_private_key_file=" + \
-                    auth_item[4]
-        elif auth_item[3] == '':
-            auth_pass_or_key = '\n'
-            if (not auth_item[4] == '') and auth_item[4]:
-                auth_pass_or_key += "ansible_ssh_private_key_file=" + \
-                    auth_item[4]
+        if (not cred_pass == '') and cred_pass:
+            vars_dict['ansible_ssh_pass'] = str_to_ascii(cred_pass)
+            if (not cred_sshkey == '') and cred_sshkey:
+                vars_dict['ansible_ssh_private_key_file'] = \
+                    str_to_ascii(cred_sshkey)
+        elif cred_pass == '':
+            if (not cred_sshkey == '') and cred_sshkey:
+                vars_dict['ansible_ssh_private_key_file'] = \
+                    str_to_ascii(cred_sshkey)
 
-        string_to_write += auth_pass_or_key
-
-        vault.dump(string_to_write, ping_inventory)
-        ping_inventory.close()
+        yml_dict = {'all': {'hosts': hosts_dict, 'vars': vars_dict}}
+        vault.dump_as_yaml_to_file(yml_dict, 'data/ping-inventory.yml')
 
         cmd_string = 'ansible all -m' \
-                     ' ping  -i data/ping-inventory --ask-vault-pass -f ' \
+                     ' ping  -i data/ping-inventory.yml --ask-vault-pass -f ' \
```

Comment:
since we are calling ansible we get the support that ansible provides. But in this case since we generated the `data/ping-inventory.yml` file here: https://github.com/quipucords/rho/blob/issues/38-part4/rho/scancommand.py#L88 
We are using the same extension.